yapapi/engine.py

```diff
@@ -412,12 +411,12 @@ async def _agreement_payment_attempt(self, agreement_id: str) -> None:
     @staticmethod
     def _check_for_termination_reason(
         activity_id: str, duration: float, num_notes: int, interval: int, payable: bool
-    ):
+    ) -> Optional[Dict[str, str]]:
         freq_descr = f"{num_notes} notes/{duration}s"
         logger.debug(
             f"{'Payable Debit notes' if payable else 'Debit notes'} for activity {activity_id}: {freq_descr}"
         )
-        if duration + DEBIT_NOTE_INTERVAL_GRACE_PERIOD < num_notes * interval:
+        if duration + DEBIT_NOTE_INTERVAL_GRACE_PERIOD < max(1, num_notes - 1) * interval:
```

Comment:
do we need this `max` here? I thought it shouldn't matter that we get a negative number on the right side... ?